core/trie.go

```diff
@@ -144,6 +146,21 @@ func NewTrie(storage TrieStorage, height uint) *Trie {
 	}
 }
 
+// DoOnTempTrie creates an in-memory Trie of height `height` and runs `do` on that Trie
+func DoOnTempTrie(height uint, do func(*Trie) error) error {
+	db, err := db.NewInMemoryDb()
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	txn := db.NewTransaction(true)
+	defer txn.Discard()
```

Comment:
Shouldn't this be `txn.Commit()`?

Reply:
And we should follow the guide for manually creating transactions as mentioned [here](https://dgraph.io/docs/badger/get-started/#read-write-transactions). Especially. if the transaction become too big?

Reply:
The DB we create here is completely temporary and wont persist beyond the scope of this function. So there is no need to go the extra mile and call `Commit()`. Using a helper like `tx.Update()` will automatically `Commit()` and we dont really need that.

`Commit()` is only necessary if you want the changes to be applied to the DB and visible to new transactions created after the `Commit()` call.

Reply:
>Since discard should only be used for read transactions

It is perfectly normal to call Discard on RW transaction. db.Update() does that as well.https://github.com/dgraph-io/badger/blob/v1.6.2/txn.go#L689